service/src/main/java/com/epam/esm/service/service/EntityService.java

```diff
@@ -86,4 +80,14 @@
     default String decodeParam(String param) {
         return URLDecoder.decode(param, StandardCharsets.UTF_8);
     }
+
+    default T checkIfPresent(Optional<T> entityOptional, String parameterName,
```

Comment:
you can just use orElseThrow(...) instead of this custom method

Reply:
I'll do it

Reply:
if the issue is with constructing NotFoundException, I can suggest to create a separate method that will populate exception with needed parameters, as logic with Optional already exists